spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServicePropertySourceLocator.java

```diff
@@ -289,6 +294,10 @@ public GenericRequestHeaderInterceptor(Map<String, String> headers) {
 			this.headers = headers;
 		}
 
+		public Map<String, String> getHeaders() {
```

Comment:
Since this is only used for testing purposes can you just extend the class in the test and add the method there?

Reply:
Are you sure downcasting like this will work , as we are creating an instance of GenericRequestHeaderInterceptor in getSecureRestTemplate method, even if we create a subclass of GenericRequestHeaderInterceptor in the test for the getter method , casting to that subclass will fail.

Reply:
In Object oriented terms, suppose the subclass is called TestRequestHeaderInterceptor then , a genericRequestHeaderInterceptor cannot be a TestRequestHeaderInterceptor but vice-versa is true.

Reply:
I was talking about creating a subclass of `ConfigServicePropertySourceLocator`

Reply:
I could be missing something, but I don't see how that would help. You will need to extend `GenericRequestHeaderInterceptor` to add the getter for the headers , as it is in the inner class. Feel free to correct me if I am wrong.